writeaheadlog_test.go

```diff
@@ -179,6 +147,63 @@ func TestCommitFailed(t *testing.T) {
 	}
 }
 
+// TestWALInconsistentSync verifies that the WAL handles the case where the
+// file randomly fails writes. When the file failes a write, it returns an
+// error on sync.
+func TestWALInconsistentSync(t *testing.T) {
+	var wt *walTester
+	err := build.Retry(50, time.Millisecond*100, func() error {
+		tester, err := newWALTester(t.Name(), faultyDiskDependency{})
+		if err != nil {
+			return err
+		}
+		wt = tester
+		return nil
+	})
```

Comment:
why do we need to retry here?

Reply:
The `faultyDiskDependency` can sometimes fail writing the metadata header and end up with a WAL that is just a blank file, which causes an error. We decided this was intended behavior, so this test just retries until it can create a WAL tester successfully.

Reply:
The WAL writes the metadata to disk during creation. I assume we need to retry in case this write fails.

Reply:
could we create the WAL first and then set the dependency?

Reply:
we could create it with a `prodDependency` and change that afterwards.

Reply:
We could, but the way the WAL is currently implemented means the file handle is opened using the production dependencies. We would need to close the file handle, and manually replace it with a `faultyFile`.

Reply:
Right, `retry` might be the cleaner solution.

Reply:
ok. But add a comment explaining why